nautobot/extras/datasources/git.py

```diff
@@ -328,9 +215,7 @@ def git_repository_dry_run(repository_record, job_result=None, logger=None):  #
 
     except Exception as exc:
         if job_result:
-            job_result.set_status(JobResultStatusChoices.STATUS_FAILURE)
             job_result.log(str(exc), level_choice=LogLevelChoices.LOG_FAILURE, logger=logger)
```

Comment:
Would be nice if `JobResult.log` calls anywhere outside of a job class can be removed.